src/compiler/checker.ts

```diff
@@ -18474,6 +18460,9 @@ namespace ts {
 
         function checkTypeReferenceNode(node: TypeReferenceNode | ExpressionWithTypeArguments) {
             checkGrammarTypeArguments(node, node.typeArguments);
+            if (node.kind === SyntaxKind.TypeReference && node.typeName.jsdocDot && !isInJavaScriptFile(node) && !findAncestor(node, n => n.kind === SyntaxKind.JSDocTypeExpression)) {
+                grammarErrorOnNode(node, Diagnostics.JSDoc_types_can_only_be_used_inside_documentation_comments);

```

Comment:
So if I write: `let x: N.M.O.<P, Q>;` I'll get a grammar error on the whole node... it would be hard to spot the extra `.`. Could we try to calculate the range between `O` and `<` for the error, or use a more specific message?